pkg/tests/ossm/deploy_on_infra_test.go

```diff
@@ -0,0 +1,183 @@
+		if !env.GetSMCPVersion().GreaterThanOrEqualTo(version.SMCP_2_3) {
+			t.T().Skip("Deploy On Infra node is available in SMCP versions v2.3+")
+		}
+		if Smcp.Rosa {
+			t.Skip("Skipping test on ROSA due to lack of permissions")
+		}
+		t.Cleanup(func() {
+			shell.Execute(t,
+				`oc adm taint nodes -l node-role.kubernetes.io/infra node-role.kubernetes.io/infra=reserved:NoSchedule- node-role.kubernetes.io/infra=reserved:NoExecute-`)
+			oc.Label(t, "", "node", workername, "node-role.kubernetes.io/infra-")
+			oc.Label(t, "", "node", workername, "node-role.kubernetes.io-")
+		})
+
+		t.LogStep("Setup: Get a worker node from the cluster that does not have the istio operator installed, label it as infra")
+		workername = pickWorkerNode(t)
+		oc.Label(t, "", "node", workername, "node-role.kubernetes.io/infra=")
+		oc.Label(t, "", "node", workername, "node-role.kubernetes.io=infra")
+		t.Log(fmt.Sprintf("Worker node selected: %s", workername))
+
+		t.LogStep("Taint node and edit the subscription to add the infra node to the node selector")
+		shell.Execute(t,
+			`oc adm taint nodes -l node-role.kubernetes.io/infra node-role.kubernetes.io/infra=reserved:NoSchedule node-role.kubernetes.io/infra=reserved:NoExecute`)
+
+		t.NewSubTest("operator").Run(func(t TestHelper) {
+			t.Log("Verify OSSM Operator is deployed on infra node when configured")
+			t.Log("Reference: https://issues.redhat.com/browse/OSSM-2342")
+			t.Cleanup(func() {
+				shell.Execute(t,
+					`oc patch subscription servicemeshoperator -n openshift-operators --type json -p='[{"op": "remove", "path": "/spec/config/tolerations"}]'`)
+				oc.WaitSMCPReady(t, meshNamespace, smcpName) // Wait for the SMCP to be ready before move to next subtest case (Because untaint)
+			})
+
+			t.LogStep("Patch subscription to run on infra nodes and wait for the operator pod to be ready")
+			oc.Patch(t, "openshift-operators", "subscription", "servicemeshoperator", "merge", `
+spec:
+  config:
+    nodeSelector:
+      node-role.kubernetes.io/infra: ""
+    tolerations:
+    - effect: NoSchedule
+      key: node-role.kubernetes.io/infra
+      value: reserved
+    - effect: NoExecute
+      key: node-role.kubernetes.io/infra
+      value: reserved
+`)
+
+			t.LogStep(fmt.Sprintf("Verify operator pod is running on the infra node. Node expected: %s", workername))
+			retry.UntilSuccess(t, func(t test.TestHelper) {
+				locator := pod.MatchingSelector("name=istio-operator", "openshift-operators")
+				operatorPod := locator(t, oc.DefaultOC)
+				oc.WaitPodReady(t, locator)
+				shell.Execute(t,
+					fmt.Sprintf(`oc get pod -n openshift-operators %s -o jsonpath='{.spec.nodeName}'`, operatorPod.Name),
+					assert.OutputContains(
+						workername,
+						"Success: Operator pod is running on the infra node",
+						"Error: Operator pod is not running on the infra node"))
+			})
+		})
+
+		t.NewSubTest("control plane").Run(func(t TestHelper) {
+			t.Log("Testing: Run OSSM Operator on infra nodes")
```

Comment:
Incorrect log statement. Thsi subtest is running the control plane on infra nodes. Also, the message should be similar to the one in line 65 (e.g. "Verify that all control plane components...")